waku/v2/protocol/waku_rln_relay/waku_rln_relay_utils.nim

```diff
@@ -386,22 +386,55 @@ when defined(rln) or (not defined(rln) and not defined(rlnzerokit)):
 
     return proofBytes
 
-  proc proofVerify*(rlnInstance: RLN[Bn256], data: openArray[byte],
-    proof: RateLimitProof): bool =
+  proc getMerkleRoot*(rlnInstance: RLN[Bn256]): MerkleNodeResult =
+    # read the Merkle Tree root after insertion
+    var
+      root {.noinit.}: Buffer = Buffer()
+      rootPtr = addr(root)
+      getRootSuccessful = getRoot(rlnInstance, rootPtr)
+    if not getRootSuccessful: 
+      return err("could not get the root")
+    if not root.len == 32:
+      return err("wrong output size")
+
+    var rootValue = cast[ptr MerkleNode] (root.`ptr`)[]
+    return ok(rootValue)
+
+  proc validateRoot*(rlnInstance: RLN[Bn256], proof: RateLimitProof): RlnRelayResult[bool] =
+    # Validate against the local merkle tree
+    let localTreeRoot = rlnInstance.getMerkleRoot()
+    if not localTreeRoot.isOk():
+      return err(localTreeRoot.error())
+    if localTreeRoot.value() == proof.merkleRoot:
+      return ok(true)
+    else:
+      trace "Local tree root does not match the root sent by peer", localTreeRoot=localTreeRoot, messageRoot=proof.merkleRoot
```

Comment:
```suggestion
      trace "Local tree root does not match the supplied proof's root", localTreeRoot=localTreeRoot, messageRoot=proof.merkleRoot
```

Reply:
Decided to remove this line altogether, in [d367f91](https://github.com/status-im/nwaku/pull/1158/commits/d367f9197738026ca13071f797de91a9ac30b3ad)